### PR TITLE
fix(yaml): quote leading-zero integer-shaped strings in stringify

### LINKED ISSUE #7040: @std/yaml zero-started numeric strings stringify to numbers
**Describe the bug**
zero-started numeric strings like '08' or '09' stringify to numbers

**Steps to Reproduce**

```js
import {stringify} from 'jsr:@std/yaml';

stringify('07'); // returns expected result "'07'\n"
stringify('08'); // returns unexpected result "08\n"
stringify('09'); // returns unexpected result "09\n"
```

**Expected behavior**

stringify('08') should return "'08'\n"
stringify('09') should return "'09'\n"

**Environment**

- OS: Debian 13
- deno version: 2.7.4 (stable, release, x86_64-unknown-linux-gnu)
- @std/yaml version: 1.0.12

### PR DESCRIPTION
**Pull request by LeSingh1**
Fixes #7040.

resolveYamlInteger accepts "07" as octal so stringify already quoted it, but "08" and "09" fell through because they aren't valid octal:

```js
import { stringify } from "jsr:@std/yaml";
stringify("07"); // "'07'\n"  ok
stringify("08"); // "08\n"    looks like a number
stringify("09"); // "09\n"    looks like a number
```

These still read as numbers to humans and to other YAML parsers, so they should be quoted too. The fix adds a `/^[-+]?0[0-9]+$/` guard in `chooseScalarStyle` so any integer-shaped string with a leading zero forces single-quoted style. After the patch:

```
"08"   -> "'08'\n"
"09"   -> "'09'\n"
"089"  -> "'089'\n"
"-08"  -> "'-08'\n"
"0a8"  -> "0a8\n"   (unchanged, not integer-shaped)
```

Tests added in `yaml/stringify_test.ts` cover the issue cases plus edge cases (signed, mixed digits, non-integer-shaped strings).

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Fix looks correct and is at the right layer (verified locally: round-trips clean, `deno test -A yaml/` 90/90, fmt+lint clean). Just two non-blocking notes inline.

File: yaml/_dumper_state.ts [hunk 1]
```diff
@@ -39,6 +39,12 @@ const STYLE_DOUBLE = 5;
 
 const LEADING_SPACE_REGEXP = /^\n* /;
 
+// Matches integer-shaped strings with a leading zero (e.g. "08", "09",
+// "089", "-012"). resolveYamlInteger rejects these when they contain a
+// non-octal digit, but they still read as numbers to humans and to other
+// YAML parsers, so force-quote them for cross-parser safety.
+const AMBIGUOUS_LEADING_ZERO_INT_REGEXP = /^[-+]?0[0-9]+$/;
+
```
Comment (bartlomieju) [inline yaml/_dumper_state.ts:47]: Non-blocking: the same cross-parser ambiguity exists for other leading-zero number shapes this regex doesn't cover. On this branch these still emit plain:

- `08.5` -> `08.5` (a YAML 1.2 parser reads this as the float `8.5` -- the direct float analogue of the bug being fixed here)
- `0o7` -> `0o7` (YAML 1.2 octal -> `7`)

These are arguably out of scope for #7040 (which is specifically leading-zero *integers*), so no need to expand the fix here -- but worth a one-line note in the PR description or a follow-up issue so it isn't assumed that *all* leading-zero numeric strings are now safe.

File: yaml/stringify_test.ts [hunk 2]
```diff
@@ -906,3 +906,24 @@ tags:
     );
   },
 });
+
+Deno.test({
+  name: "stringify() quotes leading-zero numeric strings",
```
Comment (bartlomieju) [inline yaml/stringify_test.ts:911]: Two small suggestions (optional):

1. Consider also asserting the round-trip directly, e.g. `assertEquals(parse(stringify("08")), "08")`. That's the actual contract this fix protects (verified it holds), and it guards against future parser changes.
2. Given the `08.5` / `0o7` gaps noted on the regex, the name `"quotes leading-zero numeric strings"` slightly overstates scope -- `"...leading-zero integer-shaped strings"` matches what the regex actually covers.

**Comment by tamerkalla** [comment]
I ran into this class of bug from the other direction recently and enumerated it, in case the numbers are useful for deciding how wide this PR should go.

I took 70 strings that YAML 1.1 assigns an implicit type to, ran each through a stringifier as a mapping value, then read the output back with a YAML 1.1 reader and compared it to the original string. Leading-zero integers are one family in that set; the others are the 1.1 booleans (`y`, `yes`, `no`, `on`, `off` and their case variants), sexagesimals (`12:30`, `1:2:3`, `190:20:30`), underscore-separated integers (`1_000`), binary literals (`0b101`), and unquoted timestamps (`2001-12-14`).

Results, Node 22.22.2:

- `js-yaml` 5.4.1 `dump`: 0 of 70 come back as a different value. It quotes all of the above today, including `'01234'`, `'0755'`, `'1_000'`, `'12:30'` and `'2001-12-14'`.
- `yaml` 2.9.0 `stringify`: 26 of 70 come back as a different value. Leading zeros are not among them — that library emits `01234` as a quoted string already — but the booleans, sexagesimals, underscore integers and timestamps all round-trip to `true`, `750`, `1000` and a `Date`.

Reproduction:

```js
import { stringify, parse } from "yaml";

const strings = ["y","yes","no","on","off","12:30","1:2:3","1_000","0b101","2001-12-14"];
for (const s of strings) {
  const out = stringify({ k: s });
  const back = parse(out, { schema: "yaml-1.1" }).k;
  if (back !== s) console.log(JSON.stringify(s), "->", out.trim(), "->", JSON.stringify(back));
}
```

The reason it matters beyond correctness-for-its-own-sake is that the reader is usually not the writer. A file emitted by a JS tool and read by PyYAML, Psych, or `yaml.v2` — which is what `sigs.k8s.io/yaml` sits on — is read under 1.1 rules, so every one of these families changes value at the boundary rather than at the point where anyone is looking.

If it is useful, the same enumeration in the read direction is in the tests of a package I wrote for the reverse problem: https://github.com/tamerkalla/yaml-drift
